frontend/workflows/projectSelector/src/dash.tsx

```diff
@@ -74,6 +83,16 @@ const dashReducer = (state: DashState, action: DashAction): DashState => {
   }
 };
 
+const refreshReducer = (state: RefreshRateState, action: RefreshRateAction): RefreshRateState => {
+  switch (action.type) {
+    case "UPDATE": {
+      return action.payload;
```

Comment:
Should we do some validation here against the refresh interval? e.g. it probably shouldn't be negative. Do we want to allow 0 as a value?

Reply:
hm yeah I think we want only vals greater than 0, something else I thought of as well is that it is kind of awkward to use `null` as a value denoting that we want to turn off refreshing. Maybe we could use an extra bool or something for actually turning refreshing off? idk

Reply:
yeah I think that 0 would denote no refreshing because the refresh interval would be 0 :)

Reply:
something interesting I read about was that in browsers if you nest beyond 5 setIntervals, it has a built in mechanism to make the smallest value something like 4 ms, because the idea is that 0-3 ms is too fast.

https://developer.mozilla.org/en-US/docs/Web/API/setInterval#delay_restrictions